tests/unit/testplan/common/utils/test_match.py

```diff
@@ -233,21 +240,41 @@ def test_not_match_between(self, basic_logfile):
         assert matcher.not_match_between(r"fifth", "start", "end")
         assert not matcher.not_match_between(r"third", "start", "end")
 
-    def test_get_between(self, basic_logfile):
+    @pytest.mark.parametrize("is_binary", [True, False])
+    def test_get_between(self, basic_logfile, is_binary):
         """Does the LogMatcher return the required content between marks."""
-        matcher = LogMatcher(log_path=basic_logfile)
-        matcher.match(regex=re.compile(r"second"))
-        matcher.mark("start")
-        matcher.match(regex=re.compile(r"fourth"))
-        matcher.mark("end")
-        content = matcher.get_between()
-        assert content == "first\nsecond\nthird\nfourth\nfifth\n"
-        content = matcher.get_between(None, "end")
-        assert content == "first\nsecond\nthird\nfourth\n"
-        content = matcher.get_between("start", None)
-        assert content == "third\nfourth\nfifth\n"
-        content = matcher.get_between("start", "end")
-        assert content == "third\nfourth\n"
+        matcher = LogMatcher(log_path=basic_logfile, binary=is_binary)
+        matcher.match(regex=re.compile(binary_or_string("second", is_binary)))
```

Comment:
make the binary_or_string either a fixture that depends on is_binary or a local function definition that use the is_binary from closure. You get less repetition

Reply:
Makes sense, let me see which one looks nicer when written.